pkg/broker/broker.go

```diff
@@ -101,16 +101,15 @@ func (a AnsibleBroker) getServiceInstance(instanceUUID uuid.UUID) (*apb.ServiceI
 
 }
 func (a AnsibleBroker) Login() error {
-	a.log.Debug("Retrieving serviceaccount token")
-	token, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+	clientConfig, err := k8srestclient.InClusterConfig()
 	if err != nil {
-		a.log.Debug("Error reading serviceaccount token")
+		a.log.Warning("Failed to create a restclient.InClusterConfig: %v.", err)
 		return err
 	}
 
-	return apb.OcLogin(a.log, "https://kubernetes.default",
-		"--certificate-authority", "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
-		"--token", string(token),
+	return apb.OcLogin(a.log, clientConfig.Host,
+		"--certificate-authority", clientConfig.CAFile,
+		"--token", clientConfig.BearerToken,
```

Comment:
Yes, this is using the same mechanism we use for creating the client to talk to k8s API, namely using the environment variables KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT.